Description: Fix crash when extending menu entry line beyond 79 characters
Author: Vladimir Serbinenko <phcoder@gmail.com>
Bug-Debian: http://bugs.debian.org/615893
Forwarded: yes
Applied-Upstream: http://bazaar.launchpad.net/~vcs-imports/grub/grub2-bzr/revision/3109
Last-Update: 2011-03-24

Index: b/grub-core/normal/menu_entry.c
===================================================================
--- a/grub-core/normal/menu_entry.c
+++ b/grub-core/normal/menu_entry.c
@@ -87,7 +87,7 @@
 {
   linep->len = 0;
   linep->max_len = 80; /* XXX */
-  linep->buf = grub_malloc (linep->max_len);
+  linep->buf = grub_malloc (linep->max_len + 1);
   if (! linep->buf)
     return 0;
 
